File: triage_cli/models/codex.py
"""
Codex model adapter.

Uses the Codex CLI for analysis.
"""

import os
from .base import SubprocessModel


class CodexModel(SubprocessModel):
    """Adapter for Codex (OpenAI) CLI."""

    def __init__(self):
        super().__init__()
        self.name = "codex"
        self.cmd_env_var = "TRIAGE_CODEX_CMD"
        self.default_cmd = ["codex"]

    def _build_command(self, prompt_file: str) -> list[str]:
        """Build Codex CLI command."""
        # Codex CLI: exec subcommand, --skip-git-repo-check, prompt as arg
        # Matching the working verify_system.sh pattern
        cmd_str = f"codex exec --skip-git-repo-check \"$(cat '{prompt_file}')\""
        return ["bash", "-c", cmd_str]

File: src/models/codex.ts
/**
 * Codex model adapter.
 *
 * Uses the Codex CLI for analysis.
 * Prompt is passed via stdin using `-` as the prompt argument to avoid
 * E2BIG errors when the context is large (>128KB).
 *
 * Ported from triage_cli/models/codex.py
 */

import { SubprocessModel } from './base.js';

export class CodexModel extends SubprocessModel {
  constructor() {
    super();
    this.name = 'codex';
    this.cmdEnvVar = 'TRIAGE_CODEX_CMD';
    this.defaultCmd = ['codex'];
  }

  /**
   * Build Codex CLI command.
   *
   * Uses --full-auto for non-interactive operation.
   * No sandbox restriction — Codex can explore the filesystem freely.
   * Prompt is read from stdin (base class writes it); `-` tells codex to read
   * the prompt from stdin instead of expecting a positional argument.
   */
  _buildCommand(_promptFile: string): {
    cmd: string[];
    env: Record<string, string | undefined>;
  } {
    const cmd = [...this.command, 'exec', '--full-auto', '--skip-git-repo-check', '-'];
    return {
      cmd,
      env: {},
    };
  }
}
